src/main/java/sparksoniq/spark/closures/ForClauseFlatMapClosure.java

```diff
@@ -20,36 +20,35 @@
 package sparksoniq.spark.closures;
 
 import org.apache.spark.api.java.function.FlatMapFunction;
-import org.apache.spark.api.java.function.Function;
-import org.apache.spark.api.java.function.MapFunction;
 import org.apache.spark.sql.Row;
-import org.apache.spark.sql.RowFactory;
 import org.apache.spark.sql.types.StructType;
 import sparksoniq.jsoniq.item.Item;
 import sparksoniq.jsoniq.runtime.iterator.RuntimeIterator;
 import sparksoniq.semantics.DynamicContext;
-import sparksoniq.spark.SparkSessionManager;
 
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 
 public class ForClauseFlatMapClosure implements FlatMapFunction<Row, Row> {
-    RuntimeIterator _expression;
+    private RuntimeIterator _expression;
+    private StructType _oldSchema;
```

Comment:
I suggest to use the terminology "inputSchema" and "outputSchema" to be more accurate.

Reply:
Perfect suggestion, changes made.

Reply:
Thanks!